### PR TITLE
Add bundle endpoint for ValueSets

### LINKED ISSUE #8: Add cached "Get ValueSet bundle" endpoint

### PR DESCRIPTION
**Pull request by nickclyde**
Resolves #8

Also changed the redis hostname to localhost to enable running locally outside of docker

### CODE REVIEW

**Review by akintner — APPROVED** [review]
Looks good to me 🚀

File: app/main.py [hunk 1]
```diff
@@ -11,11 +11,12 @@
 from fastapi_cache.decorator import cache
 
 from redis import asyncio as aioredis
+import requests
 
 
 @asynccontextmanager
 async def lifespan(_: FastAPI) -> AsyncIterator[None]:
-    redis = aioredis.from_url("redis://redis:6379")
+    redis = aioredis.from_url("redis://localhost:6379")
```
Comment (akintner) [inline app/main.py:19]: yay, nice tweak here.

**Review by gibuda — APPROVED** [review]

File: app/main.py [hunk 2]
```diff
@@ -38,3 +39,20 @@ async def index():
...
+):
+    url = "https://phinvads.cdc.gov/baseStu3/ValueSet"
+    params = {
+        "name": name,
+        "title": title,
+        "identifier": identifier,
+        "_getpages": _getpages,
+    }
+    response = requests.get(url, params=params)
+    return response.json()
```
Comment (katyasoup) [inline app/main.py]: Should we add in error handling now, or do we want to break that out into a separate ticket? (Seems like most errors are returned from PV as plain text, so when we hit this it throws a 500)

Comment (gibuda) [reply app/main.py]: ooo this is a good q. i did not add error handling on my open PR either, but open to doing that!

Comment (nickclyde) [reply app/main.py]: Good callout! I will add it in :)

File: app/main.py [hunk 3]
```diff
@@ -54,5 +42,16 @@ def value_set(
         "identifier": identifier,
```
Comment (katyasoup) [inline app/main.py:47]: can you add `code` and `version` params here? (I'm looking at sec. 5.1.2 and 5.1.3 in the docs for reference)

File: app/main.py [hunk 4]
```diff
@@ -54,5 +42,16 @@ def value_set(
...
         "_getpages": _getpages,
     }
-    response = requests.get(url, params=params)
-    return response.json()
+    return get(url, params)
+
+
+def get(url, params):
```
Comment (katyasoup) [inline app/main.py:55]: ❤️ this!

**Review by katyasoup — COMMENTED** [review]
looks great! added one note about additional params but otherwise happy to 👍

File: app/main.py (hunk 3 above)
Comment (nickclyde) [reply app/main.py:47]: Added, good catch!

File: app/main.py [hunk 5]
```diff
@@ -11,30 +11,54 @@
...
-def blocking():
-    time.sleep(2)
-    return {"ret": 42}
+# Get https://phinvads.cdc.gov/baseStu3/ValueSet and return response
+@app.get("/phinvads/ValueSet")
+@cache(namespace="phinvads", expire=3600)
+def value_set(
+    name: str = None,
+    title: str = None,
+    identifier: str = None,
```
Comment (gibuda) [inline app/main.py:38]: wondering if we want to keep spelling out `identifier` or if we can shorten it to `id` for brevity?

**Review by katyasoup — APPROVED** [review]
🎉

File: app/main.py (hunk 5 above)
Comment (akintner) [reply app/main.py:38]: I don't have an opinion either way, so whatever people choose is fine by me.

**Pull request merged**